porch/go.mod

```diff
@@ -32,22 +32,22 @@ require (
 	go.opentelemetry.io/otel/sdk v0.20.0
 	go.opentelemetry.io/otel/sdk/metric v0.20.0
 	go.opentelemetry.io/otel/trace v0.20.0
-	golang.org/x/mod v0.5.1
+	golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3
 	golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8
 	google.golang.org/genproto v0.0.0-20220222213610-43724f9ea8cf
 	google.golang.org/grpc v1.44.0
-	google.golang.org/protobuf v1.27.1
+	google.golang.org/protobuf v1.28.0
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.23.5
-	k8s.io/apimachinery v0.23.5
-	k8s.io/apiserver v0.23.5
-	k8s.io/client-go v0.23.5
-	k8s.io/component-base v0.23.5
+	k8s.io/api v0.24.0-rc.0
+	k8s.io/apimachinery v0.24.0-rc.0
+	k8s.io/apiserver v0.24.0-rc.0
+	k8s.io/client-go v0.24.0-rc.0
+	k8s.io/component-base v0.24.0-rc.0
```

Comment:
what drives the change to take dependency on RC build?

Reply:
Not an expert on k8s lib release versions. But the RC release contains the right version which will resolve the dependency conflicts that blocks [bumping builtin functions in Porch](https://github.com/GoogleContainerTools/kpt/pull/3028), and blocks [bumping kyaml in kpt](https://github.com/GoogleContainerTools/kpt/pull/3026/files).  

The former is needed for the next fishfooding and the RC seems the most stable version we can use. @mengqiy correct me if I'm wrong. 

More context is here: https://github.com/kubernetes/client-go/issues/1075

Reply:
We want to include a kyaml change (https://github.com/kubernetes-sigs/kustomize/pull/4568) which contains a performance improvement for both kpt and porch. This change requires https://github.com/kubernetes/kube-openapi/pull/283 which is merged after breaking change https://github.com/kubernetes/kube-openapi/pull/285.
The kube-openapi breaking change was imported to k8s after v0.24.0 (https://github.com/kubernetes/kubernetes/pull/108644). It means that NO 0.23.X k8s packages (including apimachinery and client-go) work with our desired kyaml change.

According to k8s [1.24 release schedule](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.24), 0.24.0-rc.1 is expected on 04/26 and 0.24.0 is expected on 05/03.